pkg/pgmodel/common/extension/extension.go

```diff
@@ -132,9 +132,14 @@ func MigrateExtension(conn *pgx.Conn, extName string, extSchemaName string, vali
 		return fmt.Errorf("Could not get the installed extension version: %w", err)
 	}
 
-	newVersion, ok := getNewExtensionVersion(extName, availableVersions, defaultVersion, validRange, isInstalled, currentVersion)
+	if isInstalled && !upgradeExt {
+		log.Info("msg", "skipping "+extName+" extension upgrade as upgrade extension is disabled. The current extension version is "+currentVersion.String())
+		return nil
+	}
+
+	newVersion, ok := getNewExtensionVersion(extName, availableVersions, defaultVersion, validRange, isInstalled, upgradePreRelease, currentVersion)
 	if !ok {
-		return fmt.Errorf("the extension is not available at the right version, need version: %v", rangeString)
+		return fmt.Errorf("the extension is not available at the right version, need version: %v and the default version is %s", rangeString, defaultVersion)
```

Comment:
need to add extension name to log line
```suggestion
		return fmt.Errorf("The %v extension is not available at the right version, need version: %v and the default version is %s", extName, rangeString, defaultVersion)
```